README.md

```diff
@@ -103,6 +104,18 @@ additional metrics!
 
 See the [`Documentation`](Documentation) directory for documentation of the exposed metrics.
 
+### Metrics Deprecation
+* **Pod none generic resource metrics below are marked as deprecated and will be finally removed in kube-state-metrics v1.5.0.**
```

Comment:
I don't think this is something we can do. We should definitely mark them as deprecated and create a list of deprecated metrics, but we'll need to cut a 2.0 to drop metrics I think.

Reply:
> but we'll need to cut a 2.0 to drop metrics I think.

What is the requirement for us to prepare the 2.0 release? IMHO, kube-state-metrics has no  significant none compatible changes for now.

Reply:
> kube-state-metrics has no significant none compatible changes for now.

Exactly which is why if we do introduce breaking changes, we need to cut a new major version, which I would like to avoid at this point, and keep collecting deprecated metrics for some time until we're comfortable and then cut a new major release dropping all deprecated metrics. For anyone who doesn't want deprecated metrics we can already have a flag to disable them.

Reply:
Whether dropping deprecated metrics should be considered as a none compatible breaking change is fine to me as long as we have state the deprecation clearly and leave a long enough time to let end users learn and prepare for that. So stating that deprecated metrics will be dropped in v2.0.0 release is also fine. Will do.